src/test/java/seedu/address/logic/commands/GemCommandTest.java

```diff
@@ -0,0 +1,107 @@
+    public static final String TEST_GEH_SEM1_PATH = "src/main/resources/data/GEM/GEHs1.json";
+    public static final String TEST_GEH_SEM2_PATH = "src/main/resources/data/GEM/GEHsem2.json";
+    private Model model;
+    private GemCommand gemCommand = new GemCommand();
+    private ObservableList<Module> testModules;
+    public void setUpTestModules1() throws IOException, DataConversionException {
+        JsonGradPadStorage storage = new JsonGradPadStorage(Paths.get(TEST_GEH_SEM1_PATH));
+        ReadOnlyGradPad gradPad = storage.readGradPad().get();
+        testModules = gradPad.getModuleList();
+    }
+    public void setUpTestModules2() throws IOException, DataConversionException {
+        JsonGradPadStorage storage = new JsonGradPadStorage(Paths.get(TEST_GEH_SEM2_PATH));
+        ReadOnlyGradPad gradPad = storage.readGradPad().get();
+        testModules = gradPad.getModuleList();
+    }
+    public void setUpSingleModule() throws IOException, DataConversionException {
+        JsonGradPadStorage storage = new JsonGradPadStorage(SINGLE_MODULE_PATH);
+        ReadOnlyGradPad gradPad = storage.readGradPad().get();
+        testModules = gradPad.getModuleList();
+    }
+    public void setUpCompiledModules(Path path) throws IOException, DataConversionException {
+        JsonGradPadStorage storage = new JsonGradPadStorage(path);
+        ReadOnlyGradPad gradPad = storage.readGradPad().get();
+        testModules = gradPad.getModuleList();
+    }
+    @Test
+    public void getSem1Storage_validTest() {
+        GemCommandStorage actual = gemCommand.getSem1Storage();
+        assertNull(actual);
+    }
+    @Test
+    public void getSem2Storage_validTest() {
+        GemCommandStorage actual = gemCommand.getSem2Storage();
+        assertNull(actual);
+    }
+    @Test
+    public void setSem1Storage_validTest() throws IOException, DataConversionException, IllegalValueException {
+        setUpTestModules1();
+        gemCommand.setSem1Storage();
+        GemCommandStorage storage = gemCommand.getSem1Storage();
+        ObservableList<Module> actual = storage.getGehModules();
+        assertEquals(testModules, actual);
+    }
+    @Test
+    public void setSem2Storage_validTest() throws IOException, DataConversionException, IllegalValueException {
+        setUpTestModules2();
+        gemCommand.setSem2Storage();
+        GemCommandStorage storage = gemCommand.getSem2Storage();
+        ObservableList<Module> actual = storage.getGehModules();
+        assertEquals(testModules, actual);
+    }
+
+    @Test
+    public void moduleExtractor_validTest() throws IOException, DataConversionException {
+        setUpSingleModule();
+        String expected = "\n" + MISSING_MODULE_1;
+        StringBuilder temp = gemCommand.moduleExtractor(testModules);
+        String actual = "" + temp;
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void nullModel_throwsNullPointerException() {
```

Comment:
Coding convention suggestion: perhaps we can include the method under test 😁 

```suggestion
    public void execute_nullModel_throwsNullPointerException() {
```

Reply:
Woops, must have missed this out....added in!